addons/udes_stock/tests/test_picking_type.py

```diff
@@ -630,3 +630,108 @@ def test19_enforce_with_empty_preprocessed_location(self):
+                                        confirm=True, assign=True)
+
+        strawberry_lot_picking_move_lines = strawberry_lot_picking.move_line_ids
+        strawberry_lot_picking_lots = strawberry_lot_picking_move_lines.mapped('lot_id')
+
+        # Assert strawberry lot picking has 1 move line which is linked to the strawberry lot
+        self.assertEqual(len(strawberry_lot_picking_move_lines), 1)
+        self.assertEqual(strawberry_lot_picking_lots, strawberry_lot)
+
+        strawberry_lot_picking_move_lines[0].qty_done = strawberry_qty
+
+        # Validate the strawberry lot picking which should not raise an exception as it contains
+        # only a single lot within the move lines
+        strawberry_lot_picking.action_done()
+
+        self.assertEqual(strawberry_lot_picking.state, 'done')
+
+        # Setup tangerine product info for multi lot picking
+        tangerine_product_info = {
+            'product': self.tangerine,
+            'qty': tangerine_qty,
+            'location_id': self.test_location_02.id,
+            'location_dest_id': self.test_location_01.id
+        }
+
+        # Update strawberry product info as the stock is now in test location 2
+        strawberry_product_info['location_id'] = self.test_location_02.id,
+        strawberry_product_info['location_dest_id'] = self.test_location_01.id
+
+        multi_lot_products_info = [strawberry_product_info, tangerine_product_info]
+
+        # Create, confirm and assign picking for the strawberry and tangerine lots
+        multi_lot_picking = self.create_picking(self.picking_type_internal,
+                                        products_info=multi_lot_products_info,
+                                        confirm=True, assign=True)
+
+        multi_lot_picking_move_lines = multi_lot_picking.move_line_ids
+        multi_lot_picking_lots = multi_lot_picking_move_lines.mapped('lot_id')
+
+        # Assert multi lot picking has 2 move lines with 1 each for 
+        # the strawberry and tangerine lots
+        self.assertEqual(len(multi_lot_picking_move_lines), 2)
+        self.assertEqual(multi_lot_picking_lots, (strawberry_lot | tangerine_lot))
+
+        multi_lot_picking_strawberry_move_line = multi_lot_picking_move_lines.filtered(
+            lambda ml: ml.product_id == self.strawberry
+        )
+
+        multi_lot_picking_tangerine_move_line = (
+            multi_lot_picking_move_lines - multi_lot_picking_strawberry_move_line
+        )
+
+        multi_lot_picking_strawberry_move_line.qty_done = strawberry_qty
+        multi_lot_picking_tangerine_move_line.qty_done = tangerine_qty
+
+        # Validate the multi lot picking which should raise an exception as it contains
+        # multiple lots
+        with self.assertRaises(ValidationError) as error:
+            multi_lot_picking.action_done()
+
+        expected_error_message = 'Cannot validate transfer with multiple lots.'
```

Comment:
I think assert raises allows you to specify the message.

Reply:
I was able to check the error message directly using `assertRaisesRegex` so have now used this in the new tests.